deepchecks/vision/utils/label_prediction_properties.py

```diff
@@ -0,0 +1,100 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module containing measurements for labels and predictions."""
+from typing import List
+from itertools import chain
+
+import torch
+
+from deepchecks.core.errors import DeepchecksValueError
+
+
+# Labels
+
+def _get_bbox_area(labels: List[torch.Tensor]) -> List[int]:
+    """Return a list containing the area of bboxes in batch."""
+    areas = [(label.reshape((-1, 5))[:, 4] * label.reshape((-1, 5))[:, 3]).tolist()
+             for label in labels]
+    return list(chain.from_iterable(areas))
+
+
+def _count_num_bboxes(labels: List[torch.Tensor]) -> List[int]:
+    """Return a list containing the number of bboxes in per sample batch."""
+    num_bboxes = [label.shape[0] for label in labels]
+    return num_bboxes
+
+
+def _get_samples_per_class_object_detection(labels: List[torch.Tensor]) -> List[int]:
+    """Return a list containing the classes in batch."""
+    classes = [tensor.reshape((-1, 5))[:, 0].tolist() for tensor in labels]
+    return list(chain.from_iterable(classes))
+
+
+def _get_samples_per_class_classification(labels: torch.Tensor) -> List[int]:
+    """Return a list containing the class per image in batch."""
+    return labels.tolist()
+
+
+DEFAULT_CLASSIFICATION_LABEL_PROPERTIES = [
+    {'name': 'Samples per class', 'method': _get_samples_per_class_classification, 'value_type': 'class'}
+]
+
+DEFAULT_OBJECT_DETECTION_LABEL_PROPERTIES = [
+    {'name': 'Samples per class', 'method': _get_samples_per_class_object_detection, 'value_type': 'class'},
+    {'name': 'Bounding box area (in pixels)', 'method': _get_bbox_area, 'value_type': 'numerical'},
+    {'name': 'Number of bounding boxes per image', 'method': _count_num_bboxes, 'value_type': 'numerical'},
+]
```

Comment:
I know they are different, and that it makes sense, but I REALLY hate that label properties and image properties have a different format.

Reply:
The main reason is image property are always numeric. But if we'll say "we support advance user properties for example weather (sunny/cloudy/rain/snow) then we can allow also categorical features and then we will need also there the value type